src/cartogram_info/read_csv.cpp

```diff
@@ -70,39 +70,104 @@ void CartogramInfo::read_csv(const argparse::ArgumentParser &arguments)
     // Get target area
     csv::CSVField area_field = row[area_col];
     double area;
+    std::string area_as_str = area_field.get();
+    char comma = ',';
+    char decimal = '.';
+    long unsigned int string_size = area_as_str.length();
     if (area_field.is_num()) {
       area = area_field.get<double>();
       if (area < 0.0) {
         std::cerr << "ERROR: negative area in CSV" << std::endl;
         _Exit(101);
       }
-    } else {
-      std::string area_as_str = area_field.get();
+    }
+
+    // Check for both commas and decimals
+    else if (
+      (area_as_str.find(comma) != std::string::npos) &&
+      (area_as_str.find(decimal) != std::string::npos)) {
+
+      // if commas come first, remove all commas
+      if (area_as_str.find(comma) < area_as_str.find(decimal)) {
+        area_as_str.erase(
+          std::remove(area_as_str.begin(), area_as_str.end(), comma),
+          area_as_str.end());
+        area = std::stod(area_as_str);
+      }
+
+      // Otherwise, remove decimals, then convert comma to decimal
+      else {
+        area_as_str.erase(
+          std::remove(area_as_str.begin(), area_as_str.end(), decimal),
+          area_as_str.end());
+        area_as_str[area_as_str.find(comma)] = '.';
+        area = std::stod(area_as_str);
+      }
+    }
+
+    // Check for commas only
+    else if (
+      (area_as_str.find(comma) != std::string::npos) &&
+      (area_as_str.find(decimal) == std::string::npos)) {
+      int count_commas = 0;
+      long unsigned int j;
+      for (j = 0; j < string_size; j++)
+        if ((area_as_str[j] == comma))
+          count_commas++;
+
+      // if there is only 1 comma, and there are only 2 numbers after the
+      // comma, convert to decimal
+      if (
+        (count_commas = 1) &&
+        ((area_as_str.find(comma) + 3) > (string_size))) {
+        area_as_str[area_as_str.find(comma)] = '.';
```

Comment:
This condition does not seem to work when I use `Bruxelles,"185,42",#e74c3c,BXL` for Belgium CSV.
Line 121 is making the condition assignment, use `==` instead.
The second logic is not robust. I would prefer a more straightforward way:
`if(count_commas == 1 and string_size - area_as_str.find(comma) == 3)`